rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/common/util/RMListenerProxy.java

```diff
@@ -86,6 +88,15 @@
 
     protected long counter = 0;
 
+    /** this locks are needed to synchronize access to rmInitialState */
```

Comment:
all these locks are scary. anyway I don t see where this readWriteLock has been used. Also should they be private and final ?

Reply:
I create readLock and writeLock from readWriteLock. But you are right that they should be private final